python/ray/data/_internal/execution/interfaces.py

```diff
@@ -183,28 +184,33 @@ def scale(self, f: float) -> "ExecutionResources":
         )
 
 
+@DeveloperAPI
 @dataclass
 class ExecutionOptions:
     """Common options for execution.
 
     Some options may not be supported on all executors (e.g., resource limits).
+
+    Attributes:
+        resource_limits: Set a soft limit on the resource usage during execution.
+            This is not supported in bulk execution mode.
+        locality_with_output: Set this to prefer running tasks on the same node as the
+            output node (node driving the execution). It can also be set to a list of
+            node ids to spread the outputs across those nodes.
+        preserve_order: Set this to preserve the ordering between blocks processed by
```

Comment:
could we also mention the default behavior? similar to `actor_locality_enabled` below?

```
preserve_order: Set this to preserve the ordering between blocks processed by operators under the streaming executor (off by default). The bulk executor always preserves order.
```